src/frontend/js/widgets/Dashboard/hooks/useDashboardRouter/index.tsx

```diff
@@ -38,7 +38,7 @@ export const useDashboardNavigate = () => {
         if (typeof to === 'number') {
           return navigate(to);
         }
-        return navigate(getDashboardRoutePath(to, values), routerOptions);
+        return navigate(generatePath(to, values as any), routerOptions);
```

Comment:
Maybe we could use the type expected by `generatePath` to type `values` parameter ?

Reply:
Nope we can't here as the type is inferred statically from the string, but here the values differs as it is generic ^^'

Reply:
This seems to work 😉 
```
return useMemo(
    () =>
      <Path extends number | LearnerDashboardPaths | TeacherDashboardPaths>(
        to: Path,
        params?: Parameters<typeof generatePath<Exclude<Path, number>>>[1],
        routerOptions?: NavigateOptions,
      ) => {
        if (typeof to === 'number') {
          return navigate(to);
        }
        return navigate(generatePath(to, params), routerOptions);
      },
    [],
  );
```